compose/config.py

```diff
@@ -63,12 +69,57 @@
 }
 
 
-def load(filename):
-    working_dir = os.path.dirname(filename)
-    return from_dictionary(load_yaml(filename), working_dir=working_dir, filename=filename)
+SUPPORTED_FILENAMES = [
+    'docker-compose.yml',
+    'docker-compose.yaml',
+    'fig.yml',
+    'fig.yaml',
+]
+
+
+log = logging.getLogger(__name__)
+
+
+ConfigDetails = namedtuple('ConfigDetails', 'config working_dir filename')
+
+
+def find(base_dir, filename):
+    if filename == '-':
+        return ConfigDetails(yaml.safe_load(sys.stdin), os.getcwd(), None)
+
+    filename = get_config_path(base_dir, filename)
+    return ConfigDetails(load_yaml(filename), os.path.dirname(filename), filename)
+
 
+def get_config_path(base_dir, file_path):
+    if file_path:
+        return os.path.join(base_dir, file_path)
```

Comment:
I think I'd find this bit of logic less confusing if it were moved into the function above. At that point, I think the functions' responsibilities are a bit better-separated, and they could also be renamed. Say:

```python
def resolve_config_details(base_dir, filename):
    if filename == '-':
        return ConfigDetails(yaml.safe_load(sys.stdin), os.getcwd(), None)

    if filename is None:
        file_path = find_config_file(base_dir)
    else:
        file_path = os.path.join(base_dir, filename)

    return ConfigDetails(load_yaml(file_path), os.path.dirname(file_path), file_path)


def find_config_file(base_dir):
    (candidates, path) = find_candidates_in_parent_dirs(SUPPORTED_FILENAMES, base_dir)
    # ...
```